client/executor/wasmtime/src/tests.rs

```diff
@@ -345,14 +346,14 @@ fn test_max_memory_pages(
 	precompile_runtime: bool,
 ) {
```

Comment:
Now that we've merged `max_memory_size` and `extra_heap_pages` this test is kinda confusing and would warrant a cleanup and an expansion adding more cases.

We essentially check three categories of things (based on the unique wat code blobs used here):
1) whether the instantiation succeeds with given `heap_pages` + memory section
2) whether growing the memory succeeds with given `heap_pages` + memory section
3) whether growing the memory fails with given `heap_pages` + memory_section

So could you maybe add helper function for each "category" of tests (each would wrap a whole `try_instantiate` call) and rewrite the test in this manner? Something like this:

```
let assert_instantiation_ok = |heap_pages: HeapPages, min: u32 max: Option<u32>| {
    // ...
};
let assert_grow_ok = ...;
let assert_grow_fails = ...;
// ...

assert_instantiation_ok(HeapPages::ExtraMax(1024), 64511, None);
assert_instantiation_fails(HeapPages::ExtraMax(1025), 64511, None);
assert_grow_ok(HeapPages::ExtraMax(1024 + 24), 0, None);
// ...
```

Basically so that we can easily cover every combination of `HeapPages` in every boundary scenario.

Reply:
I didn't fully get what you wanted. But I have seen what you meant on what doesn't make any sense. I removed the nonsense tests. Testing instantion in general doesn't make that much sense as we overwrite the memory definition.